### PR TITLE
friends-app

### PR DESCRIPTION
**Pull request by OlgaFrontend**
### Friends App

[Demo](https://olgafrontend.github.io/kottans-frontend/task_friends-app/) |  [Code](https://github.com/OlgaFrontend/kottans-frontend/tree/master/task_friends-app)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/OlgaDubynka/friends-app/css/main.css [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  cursor: pointer;
+  text-transform: capitalize;
+  font-size: 35px;
+}
+
+.app-header ul {
+  display: flex;
+}
+
+.app-header li {
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/friends-app/css/main.css]: Add  `:hover` and `:focus` state for interactive elements.

File: submissions/OlgaDubynka/friends-app/index.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+              <label for="male">male</label>
+              <input id="female" name="group-gender" class="input-gender input-female" type="radio" value="female" />
+              <label for="female">female</label>
+            </div>
+            <div class="search-name search-box">
+              <input class="search-by-name search-input" type="text" placeholder="Search by name..."></input>
+              <button type="button" class="search-name-btn search-btn">search</button>
+            </div>
+            <div class="search-age search-box">
+              <input class="search-by-age search-input" type="text" placeholder="Search by age..."></input>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/friends-app/index.html]: Add attribute `inputmode="numeric"` to `input` for better UX on mobile.
Read more: [https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode)

File: submissions/OlgaDubynka/friends-app/index.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          <p class="user-info__gender">Gender: <%- gender %></p>
+          <p class="user-info__location">City: <%- location %></p>	
+			</div>
+			</div>
+		</div>
+	</script>
+	<script src="https://cdn.jsdelivr.net/npm/lodash@4.17.4/lodash.min.js"></script>
+  <script src="js/main.js"></script>
+</body>
+</html>
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/friends-app/index.html]: Add empty line in the end of every file.

File: submissions/OlgaDubynka/friends-app/index.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+              <button type="button" class="btn-reset search-btn">Reset</button>
+            </div>
+          </form>
+
+        </div>
+        <div class="user-list">
+        </div>
+    </main>
+  </div>
+  <script type="text/lodash-template" id="user-card">
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/friends-app/index.html]: Try to use template literals to concatenate strings for template.

File: submissions/OlgaDubynka/friends-app/index.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+			<div class="user-card">
+        <div class="user-card__img-wrap">
+          <img  class="user-card__img" src="<%- img %>" alt="user-photo">
+        </div>
+        <div class="user-card__name"><%- name %></div>
+        <div class="user-card__btns">
+          <button class="search-btn btn-flw btn-card">follow</button>
+          <button class="search-btn btn-msg btn-card">message</button>
+        </div>
+        <div class="user-info"
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/friends-app/index.html]: Typo in the unclosed tag `<div>` generates an extra tag `<p>`.

File: submissions/OlgaDubynka/friends-app/js/main.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        renderUsers(isSortedUsers, compiled, USER_LIST);
+        break;
+      case target.value === 'male' || target.value === 'female':
+        const isFilteredUsers = users.filter(item => item.gender === target.value);
+        renderUsers(isFilteredUsers, compiled, USER_LIST);
+        break;
+    }
+  });
+
+  BTN_SEARCH_BY_NAME.addEventListener('click', function(e) {
```
Comment (A-Ostrovnyy) [inline submissions/OlgaDubynka/friends-app/js/main.js]: Why you add three same additional event listeners inside existing event listener?

**Comment by OlgaFrontend** [comment]
Thank you for review,  hope I understood right all your message)

**Comment by OlgaFrontend** [comment]
Thank for review) but I'm stuck with this, don't understand how to add focus after filter condition, all content is disappeared when I click on input name.

**Comment by A-Ostrovnyy** [comment]
Also add `:focus` state on `search` and `reset` buttons.

**Comment by OlgaFrontend** [comment]
@A-Ostrovnyy Hello) Thanks I forgot about :focus , but added it as you mentioned.

**Comment by A-Ostrovnyy** [comment]
Approved from my side.

**Comment by zonzujiro** [comment]
@OlgaFrontend, of course, I can. What help do you need?

**Comment by OlgaFrontend** [comment]
@zonzujiro Thank you!)))

**Pull request merged**